lib/ohai/plugins/ec2.rb

```diff
@@ -62,18 +43,53 @@ def has_ec2_dmi?
       if dmi[:bios][:all_records][0][:Version] =~ /amazon/
         Ohai::Log.debug("ec2 plugin: has_ec2_dmi? == true")
         true
+      else
+        Ohai::Log.debug("ec2 plugin: has_ec2_dmi? == false")
+        false
       end
     rescue NoMethodError
       Ohai::Log.debug("ec2 plugin: has_ec2_dmi? == false")
       false
     end
   end
 
+  # looks for a xen UUID that starts with ec2
+  # this gets us detection of Linux HVM and Paravirt hosts
+  def has_ec2_xen_uuid?
+    if ::File.exist?("/sys/hypervisor/uuid")
+      if ::File.read("/sys/hypervisor/uuid") =~ /^ec2/
+        Ohai::Log.debug("ec2 plugin: has_ec2_xen_uuid? == true")
+        return true
+      end
+    else
+      Ohai::Log.debug("ec2 plugin: has_ec2_xen_uuid? == false")
```

Comment:
This won't show up if the file exists and does not contain ec2. If you just remove the else line and put this at the end of the method unconditionally, it will probably do what you want since the if statement terminates with a return.